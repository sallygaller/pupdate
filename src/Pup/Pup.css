.Pup {
    text-align: center;
    display: flex;
    flex-direction: column;
    width: 75%;
    margin: 0 auto;
}

.Pup-img {
    max-width: 350px;
    margin: 0 auto;
}

.Pup ul {
    list-style: none;
    padding-left: 0px;
}

.Pup button {
    width: 50%;
    background-color: #4CAF50;
    color: white;
    padding: 14px 20px;
    margin: 15px auto;
    border: none;
    border-radius: 4px;
    cursor: pointer;
}

.Pup button:enabled:hover {
    background-color: #45a049;
}