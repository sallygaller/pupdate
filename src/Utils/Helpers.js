export default function playstyles(playstyle) {
  if (playstyle === "nervous") {
    return "I'm nervous or shy around other dogs.";
  }
  if (playstyle === "gentle") {
    return "I play gently.";
  }
  if (playstyle === "food-obsessed") {
    return "I'm food-obsessed!";
  }
  if (playstyle === "walks") {
    return "I like going on walks with my pup pals.";
  }
  if (playstyle === "parks") {
    return "I like going to dog parks with my pup pals.";
  }
  if (playstyle === "wrestling") {
    return "I like playfighting and wrestling.";
  }
  if (playstyle === "rambunctious") {
    return "I'm rambunctious and playful.";
  }
  if (playstyle === "ball-obsessed") {
    return "I love to play fetch!";
  }
}
