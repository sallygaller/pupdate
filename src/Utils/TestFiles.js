export const pupdates = [
  {
    id: 1,
    date: "2020-03-02T08:00:00.000Z",
    starttime: "09:30:00",
    endtime: "10:30:00",
    location: "Arbor Lodge Park",
    description: "Let's hang out at the park!",
    organizer: 1,
  },
  {
    id: 2,
    date: "2020-03-01T08:00:00.000Z",
    starttime: "15:30:00",
    endtime: "17:30:00",
    location: "Arbor Lodge Park",
    description: "Let's hang out at the park! I'll bring a frisbee",
    organizer: 1,
  },
];

export const pups = [
  {
    id: 1,
    name: "Eddie",
    breed: "dachshund",
    mix: true,
    age: "Adult",
    size: "M",
    nervous: true,
    rambunctious: false,
    gentle: true,
    wrestling: false,
    walks: true,
    parks: false,
    foodobsessed: true,
    ballobsessed: false,
    description:
      "Eddie can be a bit shy around other dogs at first, but he is really sweet! He loves going on long walks where there is a lot to sniff",
    date_added: "2020-12-25T09:28:32.615Z",
    owner: 1,
  },
  {
    id: 2,
    name: "Buddy",
    breed: "beagle",
    mix: true,
    age: "Adult",
    size: "L",
    nervous: false,
    rambunctious: true,
    gentle: false,
    wrestling: true,
    walks: false,
    parks: true,
    foodobsessed: false,
    ballobsessed: false,
    description:
      "Buddy can be a bit full on for some dogs, but he really just loves to play! He especially loves puppies",
    date_added: "2020-12-25T09:28:32.615Z",
    owner: 2,
  },
];
