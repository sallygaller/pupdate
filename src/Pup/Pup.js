import React from "react";
import { Link } from "react-router-dom";
import { playstyles, pupImage } from "../Utils/Helpers";
import "./Pup.css";

export default function Pup(props) {
  const pup = props.pups.find(
    ({ id }) => parseInt(id) === parseInt(props.match.params.pupId)
  );
  const playstyle = pup.playstyle;
  return (
    <div className="Pup">
      <h1>Hi, I'm {pup.name}!</h1>
      <img className="Pup-img" alt={pup.name} src={pupImage(pup.id)} />
      <p>
        Size: {pup.size}
        <br></br>
        Age: {pup.age}
      </p>
      <ul>
        {playstyle.map((p) => (
          <li>{playstyles(p)}</li>
        ))}
      </ul>
      <button>Back</button>
      {pup.id === 1 ? (
        <Link to={`/edit/pups/${pup.id}`}>
          <button>Edit</button>
        </Link>
      ) : null}
    </div>
  );
}
