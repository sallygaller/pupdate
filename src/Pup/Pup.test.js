import ReactDOM from "react-dom";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import Pup from "./Pup";
import { pups } from "../Utils/TestFiles";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <BrowserRouter>
      <Pup
        pup={pups[0]}
        match={{ params: { id: 1 }, isExact: true, path: "", url: "" }}
      />
    </BrowserRouter>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it("renders the UI as expected", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Pup
          pup={pups[0]}
          match={{ params: { id: 1 }, isExact: true, path: "", url: "" }}
        />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
