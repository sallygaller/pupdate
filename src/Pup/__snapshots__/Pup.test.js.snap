// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="Pup"
>
  <h1>
    Hi, I'm 
    
    !
  </h1>
  <img
    alt=""
    className="Pup-img"
    src="https://pupdate.s3-us-west-1.amazonaws.com/"
  />
  <p>
    Breed:
     
    
     
    <br />
    Size: 
    
    <br />
    Age: 
    
  </p>
  <p>
    
  </p>
  <ul>
    <h3>
      
      's Play Profile
    </h3>
    
    
    
    
    
    
    
    
  </ul>
  <div>
    <button
      onClick={[Function]}
    >
      Back
    </button>
  </div>
</div>
`;
